### PR TITLE
Fix multitouch and some more

### PR DESCRIPTION
**Pull request by pvuorela**
Seemed like cleanest having back that fullscreen MultiPointTouchArea on mobile version, but desktop can now just use TextRenderer's internal mouse handling.

### CODE REVIEW

**Comment by pvuorela** [comment]
Could also consider changing the UI so that tap with inactive keyboard just activates the keyboard without any key clicked. Would simplify event handling requirements and be more consistent with native keyboard if that gets supported later.

But maybe just get things working first :)

File: textrender.h [hunk 1]
```diff
@@ -96,7 +96,7 @@ class TextRender : public QQuickItem
     void setShowBufferScrollIndicator(bool s) { if(iShowBufferScrollIndicator!=s) { iShowBufferScrollIndicator=s; emit showBufferScrollIndicatorChanged(); } }
 
     Q_INVOKABLE QPointF cursorPixelPos();
-    Q_INVOKABLE QSizeF cellSize();
+    QSizeF cellSize();
```
Comment (rburchell) [inline textrender.h:99]: This is used still, in setTextRenderAttributes, so this will break won't it?

File: qml/mobile/Main.qml [hunk 2]
```diff
@@ -228,14 +228,80 @@ Item {
...
+                    y: parent.height - vkb.height
                     visible: textrender.activeFocus && visibleSetting
 
                     opacity: vkb.active ? 0.7 : 0.3
                     Behavior on opacity {
                         NumberAnimation { duration: textrender.duration; easing.type: Easing.InOutQuad }
                     }
                 }
+
+                // area to handle keyboard usage
```
Comment (rburchell) [inline qml/mobile/Main.qml:243]: might make sense to do this inside Keyboard.qml?

File: terminal.cpp [hunk 3]
```diff
@@ -67,9 +67,11 @@ Terminal::Terminal(QObject *parent)
     , iShowCursor(true)
     , iUseAltScreenBuffer(false)
     , iAppCursorKeys(false)
+    , iReplaceMode(false)
```
Comment (rburchell) [inline terminal.cpp:70]: nit: would prefer using 'm_' prefix, I've been doing that for all new (or changed) code. I should probably go on a mass style cleanup.

**Comment by rburchell** [comment]
I'm in two minds about this. I would certainly like to have it working, but I've also been going along the route of trying to keep TextRender independent of UI-specific details thus far. This is certainly not a bad implementation (and I appreciate that it will work out of the box for desktop), but it still makes me a little uncomfortable. I'll have a ponder about this for a bit today and see if I can think of any other options.

File: qml/mobile/Main.qml (hunk 2 above)
Comment (pvuorela) [reply qml/mobile/Main.qml:243]: If the area was size of keyboard, then yes, but now with current mobile UI it's handling full screen.

File: terminal.cpp (hunk 3 above)
Comment (pvuorela) [reply terminal.cpp:70]: Also prefer 'm_' more, but maybe better to do renamings separately. While at it, could reformat with astyle to get rid of inconsistencies.

File: textrender.h (hunk 1 above)
Comment (pvuorela) [reply textrender.h:99]: Used there as property.

**Comment by pvuorela** [comment]
For one thing, I think avoiding child mouse event filtering in textrenderer should make it more independent of UI on top of it :)

**Comment by rburchell** [comment]
For sure, yeah. That was a hack to get it moving. My original plan's next step was to subclass TextRender into a mobile-specific TextRender that did the child filtering. I guess we could still go that route with the new delivery bits.

**Comment by pvuorela** [comment]
Added couple commits for simplifying keyboard state handling.

**Comment by rburchell** [comment]
Thanks. I'll deal with splitting the mobile-specific bits into a subclass some other time. I can't think of any real alternative.

**Pull request merged**